lib/base/logger.cpp

```diff
@@ -195,21 +209,52 @@ void Logger::ValidateSeverity(const Lazy<String>& lvalue, const ValidationUtils&
 	}
 }
 
+void Logger::UpdateMinLogSeverity()
+{
+	auto result (LogNothing);
+
+	for (auto& logger : Logger::GetLoggers()) {
+		ObjectLock llock (logger);
+
+		if (logger->IsActive()) {
+			result = std::min(result, logger->GetMinSeverity());
+		}
+	}
+
+	if (Logger::IsConsoleLogEnabled()) {
+		result = std::min(result, Logger::GetConsoleLogSeverity());
+	}
+
+#ifdef _WIN32
+	if (Logger::IsEarlyLoggingEnabled()) {
+		result = std::min(result, LogCritical);
+	}
+#endif /* _WIN32 */
+
+	m_MinLogSeverity.store(result);
+}
```

Comment:
This has a race condition if multiple loggers change their severity at once (like it can happen with `Start()` of different loggers being called in parallel):
1. LoggerA updates its severity, enters this function and computes `result` but gets preempted before storing it to `m_MinLogSeverity`.
2. LoggerB updates its severity and completes this function.
3. LoggerA gets resumed and writes the now outdated value to `m_MinLogSeverity`.

Reply:
Would you say a static mutex for the complete UpdateMinLogSeverity() is enough to fix the race condition?

Reply:
Yes, in that case, if there are two calls to `UpdateMinLogSeverity()` in parallel, the first one may or may not see the update to the other logger, but there will for sure be another execution that takes it into account right afterwards.